service/history/replication/task_processor_manager.go

```diff
@@ -131,6 +133,7 @@ func (r *taskProcessorManagerImpl) Start() {
 
 	// Listen to cluster metadata and dynamically update replication processor for remote clusters.
 	r.listenToClusterMetadataChange()
+	go r.completeReplicationTaskLoop()
```

Comment:
Do we need to wait for the loop to exit before Stop() returns?

Reply:
No need to wait for the delete.